openhcl/update-rootfs.py

```diff
@@ -114,6 +114,15 @@ def process(temp_dir: str, underhill_path: str, kernel_path: str,
         append_file(underhill_cpio_gz_file_name, temp_file_name)
         os.unlink(temp_file_name)
 
+    if add_perf:
+        temp_file_name = os.path.join(temp_dir, "perf_tools.cpio.gz")
+        mode = '0755'
+        perf_tool_bin_path = os.path.join(kernel_path, "build", "native", "bin", os.environ["OPENHCL_KERNEL_ARCH"], "tools", "perf", "bin")
+        for f in ["perf", "trace"]:
+            create_cpio_from_file(os.path.join(perf_tool_bin_path, f), "/usr/bin/" + f, mode, temp_file_name, 'gzip')
```

Comment:
Instead of fussing with dynamically creating additional cpio archives within Python, can you just define an additional rootfs.config file with the files you need and pass that to this (unmodified) tool?

Reply:
(At some point recently we made it easy to provide multiple rootfs.config files.)